tls/s2n_config.c

```diff
@@ -96,10 +96,6 @@ static int s2n_config_init(struct s2n_config *config)
     config->encrypt_decrypt_key_lifetime_in_nanos = S2N_TICKET_ENCRYPT_DECRYPT_KEY_LIFETIME_IN_NANOS;
     config->decrypt_key_lifetime_in_nanos = S2N_TICKET_DECRYPT_KEY_LIFETIME_IN_NANOS;
     config->async_pkey_validation_mode = S2N_ASYNC_PKEY_VALIDATION_FAST;
-
-    /* By default, only the client will authenticate the Server's Certificate. The Server does not request or
-     * authenticate any client certificates. */
-    config->client_cert_auth_type = S2N_CERT_AUTH_NONE;
```

Comment:
I considered that, but I'm not a fan. The issues I see:
* s2n_connection_get_client_auth_type and s2n_cert_auth_type are public, so yes that could be a breaking change
* You shouldn't be able to call set_client_auth_type with S2N_CERT_AUTH_UNDEFINED, that seems strange
* We shouldn't have to handle S2N_CERT_AUTH_UNDEFINED everywhere we handle s2n_cert_auth_type. It's not a valid state.

You could fix those with separate enums that you translate between, where only one has S2N_CERT_AUTH_UNDEFINED. But we've had issues with closely related but different enums before, and I think that's a worse solution than just using the bitflag. Of course, I couldn't actually use a bitflag because of SAW, but that was the goal.

Reply:
Okay yeah, makes sense.